src/imitation/algorithms/mce_irl.py

```diff
@@ -381,9 +397,9 @@ def set_demonstrations(self, demonstrations: MCEDemonstrations) -> None:
         # Demonstrations are either trajectories or transitions;
         # we must compute occupancy measure from this.
         if isinstance(demonstrations, Iterable):
-            first_item, demonstrations = util.get_first_iter_element(demonstrations)
+            first_item, demonstrations_it = util.get_first_iter_element(demonstrations)
```

Comment:
I'm not sure how this type is incompatible since we are narrowing the scope of the type, we should be able to use the same variable if the generic is well constructed. (right?)

Reply:
I presume mypy doesn't actually narrow the scope of the variable, even if you use it like this. Not sure though